identity/client/src/pages/groups/detail/mapping-rules/AssignMappingRulesModal.tsx

```diff
@@ -147,7 +156,9 @@ const AssignMappingRulesModal: FC<
       )}
       <DropdownSearch
         autoFocus
-        items={unassignedMappingRules}
+        onChange={handleMappingRuleFilterChange}
+        items={mappingRuleSearchResults?.items || []}
+        filter={unassignedFilter}
```

Comment:
❓ Why are we passing a `filter` function to the `DropdownSearch` component instead of pre-filtering the items and passing only the relevant ones? 

It feels like the filtering logic for thing like unassigned mapping rules belongs more to the parent component. Arguably, the `DropdownSearch` should just render what it receives, rather than take on responsibility for context-specific filtering.

Was there a particular reason this filtering was delegated to the child component?

Reply:
We do want the accumulation behavior (as described above). IMO, it is only a code style decision if this behavior is implemented in the DropdownSearch component or through a custom hook.

I decided for filter + accumulation within the DropdownSearch component instead of the parent, as this is currently the only place where we use this behavior, so it makes more sense to me to implement it locally, even if the Component needs internal state with this approach

Reply:
As discussed in the sync meeting: 

Due to the accumulation behavior required by missing complex backend queries, let's keep this as-is but add documentation in the form of comments